server/src/main/java/io/crate/analyze/Analyzer.java

```diff
@@ -778,18 +779,7 @@ public AnalyzedStatement visitCreateServer(CreateServer createServer, Analysis c
         @Override
         public AnalyzedStatement visitCreateForeignTable(CreateForeignTable createForeignTable,
                                                          Analysis context) {
-            RelationName tableName = RelationName.of(
-                createForeignTable.name(),
-                context.sessionSettings().searchPath().currentSchema()
-            );
-            tableName.ensureValidForRelationCreation();
-            var tableElementsAnalyzer = new TableElementsAnalyzer(
-                tableName,
-                context.transactionContext(),
-                nodeCtx,
-                context.paramTypeHints()
-            );
-            return tableElementsAnalyzer.analyze(createForeignTable);
+            return FdwAnalyzer.analyze(context, nodeCtx, createForeignTable);
```

Comment:
Moved the logic into a static method to make it accessible for the `SQLExecutor.Builder`.

I might follow up on this and look into changing the `SQLExecutor` to not have the builder pattern like it is. That would allow a `.addForeignTable` to use the full analyzer.
And I'm not sure there's a good reason to have the split, we can (and do) change the cluster state after the `.build()`, so we could expose all the `.add` methods on the final object.

Reply:
> Depends a bit on if there's still a good reason to have the builder split, we can (and do) change the cluster state after the .build(), so we could expose all the .add methods on the final object.

I would vote for `.add` method.